egs/wsj/s5/steps/data/augment_data_dir.py

```diff
@@ -83,8 +84,8 @@ def AugmentWav(utt, wav, dur, fg_snr_opts, bg_snr_opts, fg_noise_utts, \
         num = random.choice(num_opts)
         for i in range(0, num):
             noise_utt = random.choice(bg_noise_utts)
-            noise = noise_wavs[noise_utt] + " wav-reverberate --duration=" \
-            + dur_str + " - - |"
+            noise = "wav-reverberate --duration=" \
```

Comment:
This fix doesn't seem right- it looks like you didn't undertsand what I was saying about the pipe at the end.  There needs to be some conditional code.  Read my responses to you on github and/or kaldi-help more carefully.

Reply:
Oh, I see. The noise file may be a pipline too.

Reply:
I think I can add "" to the pipeline, the result will be like this:
wav-reverberate --duration=4.588438 "wav-copy /data4/wavdata20171208/music_noise_16K_norm/norm_238.wav - |" - 
I think it works.

Reply:
My point was that the original file in the wav.scp may already have been a pipeline.

Reply:
I think it is tackled between line 111 to 118.

Reply:
Lines 111-118 handle that for the speech file you're adding noise to. You need to do the same kind of thing for the noise file too. I think what you have will fail if the noise wav.scp uses pipes.

Reply:
I add quotes to the noise wav. Now it generates like this:
wav-reverberate --duration=4.588438 "wav-copy /data4/wavdata20171208/music_noise_16K_norm/norm_238.wav - |" -
It can deal with pipeline case and non-pipeline case

Reply:
@by2101, could you also apply this fix to the foreground noises in the next loop? It looks like the fix has only been done for background noises so far. After that, I think this will be basically ready to merge.

Reply:
Well, in foreground noise loop, it works well.
The bug is caused by using wav-reberberate to change duration: 
noise = noise_wavs[noise_utt] + " wav-reverberate --duration=" + dur_str + " - - |"
when noise_wavs[noise_utt] is not a pipeline, it fail.
But in foreground, noise = noise_wavs[noise_utt] works well whenever it is pipeline or a file.